packages/stableswap-sdk/src/entities/exchange.ts

```diff
@@ -187,3 +188,47 @@ export const makeExchange = ({
   };
   return exchange;
 };
+
+/**
+ * Get exchange info from just the swap account.
+ * @param connection
+ * @param swapAccount
+ * @param tokenA
+ * @param tokenB
+ * @returns
+ */
+ export const loadExchangeInfoFromSwapAccount = async (
+  connection: Connection,
+  swapAccount: PublicKey,
+  tokenA: Token | undefined = undefined,
+  tokenB: Token | undefined = undefined
+): Promise<IExchangeInfo> => {
+  // Consider moving this makeToken function to @saberhq/token-utils.
+  const makeToken = async (address: PublicKey): Promise<Token> => {
+    const tokenAccountInfo = await connection.getAccountInfo(swap.tokenA.mint);
+    if (!tokenAccountInfo) {
+      throw new Error("Token Account Info not found.");
+    }
+    const mintInfo = deserializeMint(tokenAccountInfo.data);
+    return Token.fromMint(address, mintInfo.decimals);
+  };
+
+  const swapAccountInfo = await connection.getAccountInfo(swapAccount);
```

Comment:
Let's load the swap with this method here:
https://github.com/saber-hq/saber-common/blob/ac4d9ea9bffc61bbe104ac98b86d75ab1c856204/packages/stableswap-sdk/src/stable-swap.ts#L40